packages/webpack-index-html-plugin/webpack-index-html-plugin.js

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable class-methods-use-this, no-param-reassign */
+const path = require('path');
+const deepmerge = require('deepmerge');
+const { createEntrypoints } = require('./src/create-entrypoints');
+const { getLegacyEntries } = require('./src/get-legacy-entries');
+const { emitIndexHTML } = require('./src/emit-index-html');
+const { PLUGIN_NAME, createError } = require('./src/utils');
+
+/**
+ * @typedef {object} VariationConfig
+ * @property {string} name the name of this variation
+ * @property {string} [sharedEntry] if an entry should be shared, this variation
+ *   will not create a new entrypoint but share it with another variation
+ */
+
+/**
+ * @typedef {object} MultiIndexConfig
+ * @property {string} [fallback]
+ * @property {VariationConfig[]} variations
+ * @property {(index: string, variation: string, fallback: boolean) => string} [transformIndex]
+ */
+
+/**
+ * @typedef {object} TemplateData
+ * @property {object[]} assets
+ * @property {string[]} entries
+ * @property {string[]} legacyEntries[]
+ * @property {string} variation
+ */
+
+/**
+ * @typedef {object} WebpackIndexHTMLPluginConfig
+ * @property {(data: TemplateData) => string} template
+ * @property {MultiIndexConfig} [multiIndex]
+ * @property {boolean} [legacy]
+ * @property {boolean} [multiBuild]
+ * @property {boolean} [inject]
+ * @property {string} [legacyDir]
+ * @property {number} [legacyTimeout]
+ * @property {import('@open-wc/building-utils/index-html/create-index-html').PolyfillsConfig} [polyfills]
+ * @property {false|object} [minify] minify configuration, or false to disable minification
+ */
+
+const defaultConfig = {
+  legacyDir: 'legacy',
+  inject: true,
+  legacyTimeout: 1000 * 60 * 10,
+};
+
+class WebpackIndexHTMLPlugin {
+  /**
+   * @param {Partial<WebpackIndexHTMLPluginConfig>} config
+   */
+  constructor(config = {}) {
+    this._config = deepmerge(defaultConfig, config);
+
+    if (config.multiIndex) {
+      if (!Array.isArray(config.multiIndex.variations)) {
+        throw createError('Missing an array of variations at multiIndex.variations.');
+      }
+
+      if (config.multiIndex.variations.some(v => !v.name)) {
+        throw createError(`Some variations are missing a name property.`);
+      }
+    }
+  }
+
+  apply(compiler) {
+    // @ts-ignore
```

Comment:
added this ignore... is this if needed? (it is nowhere set... does it comes from webpack directly?)

Reply:
Yes apply comes from webpack